src/core/matcher.rkt

```diff
@@ -143,16 +143,23 @@
    [else
     (timeline-push! 'method "batch-egg-rewrite")
     (timeline-push! 'inputs (map ~a exprs))
-    (define out (batch-egg-rewrite exprs ctx #:rules rules #:depths depths))
+    ; (define out (batch-egg-rewrite exprs ctx #:rules rules #:roots root-locs #:depths depths))
+    (define e-input (make-egg-descriptor exprs rules #t #:node-limit (*node-limit*)))
+    (define p-input '())
+    (match-define (cons variantses _) (run-egg e-input p-input #t))
+    (define out
+      (for/list ([expr exprs] [variants variantses])
+        (for/list ([variant (remove-duplicates variants)])
```

Comment:
I am not sure why we need a second `remove-duplicates` here; it's it already in the `egg-herbie.rkt` code?

Reply:
I think it wasn't working for some reason when we were not doing it in one place. We can remove the one here if it will look like a better optimization

Reply:
Looks like it works without it too. Maybe I was thinking about something else!